### DISCUSSION TITLE
Error while creating cooking in server action

### DISCUSSION THREAD

**Discussion by junaid52** [vercel/next.js#60512] [Help] [answered]
Hi friends i am new on Nextjs i try to create cookie in server action, i want to store accessToken and refreshToken in cookie so i pass object to cookie value but it generates this error Error: (intermediate value)(intermediate value)(intermediate value).toUTCString is not a function

```
'use server';
import { cookies } from 'next/headers';
const createCookie = async (data) => {
  cookies().set('token', JSON.stringify(data), {
    secure: true,
    httpOnly: true,

    expires: data.expiresIn,
  });
};
export const signUp = async (formData) => {
  const data = {
    email: formData.get('email'),
    password: formData.get('password'),
    returnSecureToken: true,
  };
  const url =
    'https://identitytoolkit.googleapis.com/v1/accounts:signUp?key=something';
  const res = await fetch(url, {
    method: 'post',
    headers: {
      'content-type': 'application/json',
    },
    body: JSON.stringify(data),
  });
  const resData = await res.json();

  createCookie(resData);
  return resData;
};
```

**Comment by icyJoseph** [comment, marked as answer]
Could it be that `expiresIn` doesn't exist? Could you verify the shape of `resData`?

**Reply by junaid52** [reply]
thanks for your answer there is mistake in expiresIn sorry, and can you tell i am using this pattern for auth is it valid ?

**Reply by icyJoseph** [reply]
Auth is always a bit tricky. Relying only on a cookie to authenticate, leads the door open to impersonation, should one be able to get a hold of that cookie. One should never make assumptions about how these cookies are handled by users: 

- https://portswigger.net/web-security/csrf

You should really choose the authentication, and authorization methods, depending on how important this transaction is, and how much you are willing to risk. Reinventing auth, or doing it yourself, is often not recommended, and there's standards and such that companies follow to do proper auth-ing. 

- https://next-auth.js.org/getting-started/introduction
